api/api_ethereum.go

```diff
@@ -502,52 +503,218 @@ type EthRPCTransaction struct {
+
+	return rawTx
 }
 
 // GetTransactionCount returns the number of transactions the given address has sent for the given block number.
 func (api *EthereumAPI) GetTransactionCount(ctx context.Context, address common.Address, blockNrOrHash rpc.BlockNumberOrHash) (*hexutil.Uint64, error) {
-	// TODO-Klaytn: Not implemented yet.
-	return nil, nil
+	return api.publicTransactionPoolAPI.GetTransactionCount(ctx, address, blockNrOrHash)
 }
 
 // GetTransactionByHash returns the transaction for the given hash.
 func (api *EthereumAPI) GetTransactionByHash(ctx context.Context, hash common.Hash) (*EthRPCTransaction, error) {
-	// TODO-Klaytn: Not implemented yet.
+	// Try to return an already finalized transaction
+	if tx, blockHash, blockNumber, index := api.publicTransactionPoolAPI.b.ChainDB().ReadTxAndLookupInfo(hash); tx != nil {
+		return newEthRPCTransaction(tx, blockHash, blockNumber, index)
+	}
+	// No finalized transaction, try to retrieve it from the pool
+	if tx := api.publicTransactionPoolAPI.b.GetPoolTransaction(hash); tx != nil {
+		return newEthRPCPendingTransaction(tx)
+	}
+	// Transaction unknown, return as such
 	return nil, nil
 }
 
 // GetRawTransactionByHash returns the bytes of the transaction for the given hash.
 func (api *EthereumAPI) GetRawTransactionByHash(ctx context.Context, hash common.Hash) (hexutil.Bytes, error) {
-	// TODO-Klaytn: Not implemented yet.
-	return nil, nil
+	rawTx, err := api.publicTransactionPoolAPI.GetRawTransactionByHash(ctx, hash)
+	if rawTx == nil || err != nil {
+		return nil, err
+	}
+
+	return rawTx, nil
 }
 
 // GetTransactionReceipt returns the transaction receipt for the given transaction hash.
 func (api *EthereumAPI) GetTransactionReceipt(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
-	// TODO-Klaytn: Not implemented yet.
-	return nil, nil
+	// Formats return Klaytn Transaction Receipt to the Ethereum Transaction Receipt.
+	tx, blockHash, blockNumber, index, receipt := api.publicTransactionPoolAPI.b.GetTxLookupInfoAndReceipt(ctx, hash)
+
+	receipts := api.publicTransactionPoolAPI.b.GetBlockReceipts(ctx, blockHash)
+	cumulativeGasUsed := uint64(0)
+	for i := uint64(0); i <= index; i++ {
+		cumulativeGasUsed += receipts[i].GasUsed
+	}
+
+	ethTx, err := newEthTransactionReceipt(tx, blockHash, blockNumber, index, cumulativeGasUsed, receipt)
+	if ethTx == nil || err != nil {
+		return nil, err
+	}
+	return ethTx, nil
+}
+
+// newEthTransactionReceipt creates a transaction receipt in Ethereum format.
+func newEthTransactionReceipt(tx *types.Transaction, blockHash common.Hash, blockNumber, index, cumulativeGasUsed uint64, receipt *types.Receipt) (map[string]interface{}, error) {
+	// When an unknown transaction receipt is requested through rpc call,
+	// nil is returned by Klaytn API, and it is handled.
+	if tx == nil || receipt == nil {
+		return nil, nil
+	}
+
+	from := getFrom(tx)
+
+	// If to is nil, it is fills with from.
+	to := tx.To()
+	if to == nil {
+		to = &from
+	}
```

Comment:
Same here

Reply:
When the transaction type is SmartContractDeploy, we have to leave nil as it is.
So additional tx type checking logic should be appended.
(should consider FeeDelegation and PartialFeeDelegation also)

When the type of Tx is SmartContractDeploy transaction. | => Use nil value as it is.
When the type of Tx is AccountUpdate transaction . | => Use from value instead of nil.
When the type of Tx is Cancel transaction. | => Use from value instead of nil.
When the type of Tx is ChainDataAnchoring transaction. | => Use from value instead of nil.